doc/webdav-intro.md

```diff
@@ -0,0 +1,281 @@
+RFC 4918.
+
+Properties
+----------
+
+In `PROPFIND` requests & responses, DAV resource properties are represented by
+XML elements of the same name in the "`DAV:`" namespace (but see "XML
+Extensibility" below), with the contents of the elements (when present) giving
+the properties' values.
+
+The properties defined by the RFC that do not involve locks are as follows.
+Unless otherwise specified, the contents of the corresponding XML elements are
+`#PCDATA`.  Properties that mirror HTTP response headers use the headers'
+values as would be returned for a `GET` request without any `Accept` headers,
+and such properties must be defined if the respective headers are present in a
+`GET` response.
+
+- `creationdate` — timestamp of when the resource was created, in RFC 3339
+  format
+
+- `displayname` — name of the resource to display to the user
+
+- `getcontentlanguage` — `Content-Language` HTTP header value for the resource
+
+- `getcontentlength` — `Content-Length` HTTP header value for the resource
+
+- `getcontenttype` — `Content-Type` HTTP header value for the resource
+
+- `getetag` — `Etag` HTTP header value for the resource
+
+- `getlastmodified` — `Last-Modified` HTTP header value for the resource
+
+    - Note that the RFC states that the value of the `getlastmodified` property
+      must be in rfc1123-date format as defined by RFC 2616, §3.3.1, which, in
+      `strftime` format, is `%a, %d %b %Y %H:%M:%S GMT`, e.g., "Mon, 08 Jul
+      2024 19:05:32 GMT".
+
+- `resourcetype` — Specifies the nature of the resource.  This property must be
+  defined for all resources.  In XML, the "resourcetype" element contains zero
+  or more child elements, each of which is an identifier for a resource type
+  that the resource belongs to.  The only resource type defined by the RFC is
+  "collection", represented by an empty element of that name.  Thus, if a
+  "resourcetype" element contains a "collection" element, the resource is a
+  collection, and if it does not contain a "collection" element, the resource
+  is a non-collection.
+
+Except where otherwise noted, a resource need not have all properties defined
+on it.  If a property is not defined, it will not be listed in responses to
+"allprop" or "propname" requests, and if it is requested in a "prop" request,
+the property will be returned as an empty element in a "propstat" with a status
+of 404.
+
+
+XML Extensibility
+-----------------
+
+Custom properties and other elements can be added to WebDAV request & response
+bodies by placing them in an XML namespace other than "`DAV:`" (which is
+reserved for elements defined by RFCs).  If a server or client receives a
+document containing elements it does not recognize, it must ignore those
+elements and all of their children, except when a client attempts to set one or
+more unrecognized dead properties on a resource, in which case all of the
+properties must be recorded.
```

Comment:
Are there any good examples of such extensions?

Reply:
There are a number of RFCs that extend WebDAV (listed on the left of http://webdav.org under "Specifications"), though I expect they all use the `DAV:` namespace.